CONVENTIONS.md

```diff
@@ -0,0 +1,95 @@
+## Plugin Naming Conventions
+
+These are _recommended_ plugin naming conventions. YMMV.
+
+- [GitHub](#github)
+    - [OpenSearch Plugins](#opensearch-plugins)
+    - [OpenSearch Dashboard Plugins](#opensearch-dashboard-plugins)
+- [Metadata](#metadata)
+    - [OpenSearch Plugins](#opensearch-plugins)
+    - [OpenSearch Dashboard Plugins](#opensearch-dashboard-plugins)
+- [Artifacts](#artifacts)
+- [Folders](#folders)
+- [Classes](#classes)
+- [APIs](#apis)
+- [Indices](#indices)
+- [Identifiers](#identifiers)
+- [Variables](#variables)
+
+### GitHub
+
+- Within the [opensearch-project org](https://github.com/opensearch-project), do not include the word `OpenSearch` or `OpenSearch Dashboards` in the repo name.
+- Inherit templates, CoC, etc., from [opensearch-project/.github](https://github.com/opensearch-project/.github).
+- Use lowercase repo names.
+- Provide a meaningful description, e.g. `An OpenSearch Dashboards plugin to perform real-time and historical anomaly detection on OpenSearch data.`
+
+#### OpenSearch Plugins
+
+- Do not include the word `plugin` in the repo name, e.g. [job-scheduler](https://github.com/opensearch-project/job-scheduler).
+
+#### OpenSearch Dashboard Plugins
+
+- Prefix OpenSearch Dashboards plugins with `dashboards-`, e.g. [dashboards-reports](https://github.com/opensearch-project/dashboards-reports).
+
+### Metadata
+
+#### OpenSearch Plugins
+
+- Plugin name starts with `opensearch-`.
+- Include `opensearch-` in the plugin name.
+- Description does not end with a period.
+- Include `OpenSearch` and `plugin` in the descripion. 
+
+```groovy
+opensearchplugin {
+    name 'opensearch-job-scheduler'
+    description 'OpenSearch Job Scheduler plugin'
+    classname 'com.amazon.opensearch.jobscheduler.JobSchedulerPlugin'
+}
+```
+
+#### OpenSearch Dashboard Plugins
+
+- Plugin ID is lowercase camelcase, e.g. `notebooksDashboards`.
+- Version follows [semver](https://semver.org/).
+
+```json
+{
+  "id": "notebooksDashboards",
+  "version": "1.0.0.0-beta1",
+  "opensearchDashboardsVersion": "1.0.0-beta1",
+  "server": true,
+  "ui": true,
+  "requiredPlugins": ["navigation", "embeddable", "dashboard"],
+  "optionalPlugins": []
+}
+```
+
+### Artifacts
+
+- Artifacts are of the `<package name>-<version>` format, e.g. `opensearch-job-scheduler-1.0.0.0-beta1.zip`.
+
+### Folders
+
+- Always use lowercase.
+
+### Classes
+
+- Lowercase namespaces, e.g. `com.amazon.opensearch.jobscheduler`.
+- CamelCase classes, e.g. `com.amazon.opensearch.jobscheduler.JobSchedulerPlugin`.
+
+### APIs
+
+TODO
```

Comment:
We have decided it will be `_plugins`.
Can we add it here?
Ref: https://github.com/opensearch-project/opensearch-plugins/issues/13

Same with indices.

Reply:
Done.